### PR TITLE
EZP-26719 : Implement pjax view matcher

### PR DESCRIPTION
**Pull request by nicolas-bastien**
https://jira.ez.no/browse/EZP-26719

## Description

Right now content link on selectroot fielddescription render a front url.
This PR implement a matcher to remove the link on pjax pages so only the name is displayed

## Test

- unit test
- manually

## QA

    composer require ezsystems/platform-ui-bundle:dev-EZP-26719-implement-viewpjaxmatcher-tofix-whitepagelink

### CODE REVIEW

**Comment by nicolas-bastien** [comment]
@bdunogier @dpobel ready for code review

File: Tests/Matcher/ContentBased/PjaxTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    protected function setUp()
+    {
+        $this->matcher = new Pjax();
+    }
+
+    public function testMatch()
+    {
+        $viewMock = $this->getMockBuilder('eZ\Publish\Core\MVC\Symfony\View\View')->getMock();
+        $requestStackMock = $this->getMockBuilder('Symfony\Component\HttpFoundation\RequestStack')->getMock();
```
Comment (bdunogier) [inline Tests/Matcher/ContentBased/PjaxTest.php]: I don't think I would mock the RequestStack, I'd just build one (I think it's what the Sf tests mostly do):

```php
$pjaxRequest = new Request();
$pjaxRequest->headers->set('x-pjax', 'true');

$requestStack = new RequestStack();
$requestStack->push(new Request());
$requestStack->push($pjaxRequest);
```

File: Matcher/ContentBased/Pjax.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * This file is part of the eZ PlatformUI package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\Matcher\ContentBased;
```
Comment (bdunogier) [inline Matcher/ContentBased/Pjax.php]: I would expect the matcher to be in `PlatformUIBundle\Matcher`, as it is not Content Based. Also, I'd name it `PjaxRequest`, as it matches a, well, pjax requests :)

File: Resources/config/kernel.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+system:
+    default:
+        content_view:
+            line:
+                pjax_location_line:
+                    template: "eZPlatformUIBundle:content:line.html.twig"
```
Comment (bdunogier) [inline Resources/config/kernel.yml]: Given what this template does, I'd specialize it's name a little bit. Maybe `pjax_line.html.twig` ?

File: Resources/config/kernel.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+system:
+    default:
+        content_view:
+            line:
+                pjax_location_line:
```
Comment (bdunogier) [inline Resources/config/kernel.yml]: `pjax_line` ? It doesn't have much to do with locations.

**Comment by nicolas-bastien** [comment]
@bdunogier done

File: DependencyInjection/EzPlatformUIExtension.php [hunk 5]
```diff
@@ -45,6 +45,11 @@ public function prepend(ContainerBuilder $container)
         $this->prependYui($container);
         $this->prependCss($container);
         $this->prependImageVariations($container);
+
+        $configFile = __DIR__ . '/../Resources/config/kernel.yml';
```
Comment (bdunogier) [inline DependencyInjection/EzPlatformUIExtension.php]: I don't think I'd name this file `kernel.yml`. The standard for defining views is `views.yml`, could you rename it ?

File: Matcher/PjaxRequest.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\Matcher;
+
+use eZ\Publish\Core\MVC\Symfony\Matcher\ViewMatcherInterface;
+use eZ\Publish\Core\MVC\Symfony\RequestStackAware;
+use eZ\Publish\Core\MVC\Symfony\View\View;
+use EzSystems\PlatformUIBundle\Http\PjaxRequestMatcher;
+
+class PjaxRequest implements ViewMatcherInterface
```
Comment (bdunogier) [inline Matcher/PjaxRequest.php]: Could you please add one sentence describing what the matcher does ?

File: Resources/views/content/line.html.twig [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{% set content_name=ez_content_name(content) %}
+
+<p>{{ content_name }}</p>
```
Comment (bdunogier) [inline Resources/views/content/line.html.twig]: Any reason why we don't just call `{{ ez_content_name(content) }}` ?

File: Tests/Matcher/PjaxRequestTest.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+     * @var \EzSystems\PlatformUIBundle\Matcher\PjaxRequest
+     */
+    private $matcher;
+
+    protected function setUp()
+    {
+        $this->matcher = new PjaxRequest();
+    }
+
+    public function testMatch()
```
Comment (bdunogier) [inline Tests/Matcher/PjaxRequestTest.php]: I would have done two tests, `testMatch`, and `testNoMatch`.

File: Resources/views/content/line.html.twig (hunk 7 above)
Comment (nicolas-bastien) [inline Resources/views/content/line.html.twig]: I just took the front template code so if this is better i can change

File: DependencyInjection/EzPlatformUIExtension.php (hunk 5 above)
Comment (bdunogier) [inline DependencyInjection/EzPlatformUIExtension.php]: `views.yml` is still better, I think. Given that the kernel is meant to be split into packages in the near future, referencing the "kernel" will become a bit meaningless. Furthermore, it is what is used as a best practice (demos, tutorials...).

File: Resources/views/content/line.html.twig (hunk 7 above)
Comment (bdunogier) [inline Resources/views/content/line.html.twig]: I don't know if it's better. But I know that if we don't need it, then we can remove it :)

**Comment by nicolas-bastien** [comment]
@bdunogier done

**Comment by bdunogier** [comment]
Sorry I didn't find this earlier, but could you make this two commits ? One is the addition of the view matcher, and the other is the usage of this matcher.

**Comment by nicolas-bastien** [comment]
@bdunogier done + rebase the branch on master to get the content view to ease QA validation

**Comment by bdunogier** [comment]
Approved. Let's send this to QA.

**Comment by bdunogier** [comment]
Okay, let's merge it and have it validated during certification.

**Pull request merged**